### PR TITLE
[PORT] Table Changes From Cit (Mostly)

### PR DESCRIPTION
**Pull request by MCterra10**
## About The Pull Request
You can now make plasmaglass tables. They don't break when you stand on em and have an awesome pale-purple aesthetic.
When standing on tables your character is now shifted up by a few pixels.
You can now bang tables by alt-clicking. Harm intent to do a hard ace-attorney slam with sound effects to go with it (the lawyer update we never knew we needed).

This stuff is all stolen from various places on cit and edited for aucode.

## Why It's Good For The Game
It just is.

## Changelog
:cl:
add: Alt-click to bang tables
add: Plasmaglass tables
tweak: Characters shift up a bit when on tables
/:cl:

### CODE REVIEW

**Comment by TheFakeElon** [comment]
https://www.youtube.com/watch?v=vDMwDT6BhhE&list=LLPg6YF07x1azClymAjstqjQ&index=608&ab_channel=iteachvader

File: austation/code/game/objects/structures/tables_racks.dm [hunk 1]
```diff
@@ -0,0 +1,67 @@
...
+	if(user && Adjacent(user) && !user.incapacitated())
+		if(istype(user) && user.a_intent == INTENT_HARM)
+			user.visible_message("<span class='warning'>[user] slams [user.p_their()] palms down on [src].</span>", "<span class='warning'>You slam your palms down on [src].</span>")
+			playsound(src, 'austation/sound/misc/tableslap.ogg', 50, 1)
+			if(prob(5) && istype(src, /obj/structure/table/glass) && isliving(user))
+				var/mob/living/L = user
+				var/obj/structure/table/glass/G = src
+				G.table_shatter(L)
+		else
+			user.visible_message("<span class='notice'>[user] slaps [user.p_their()] hands on [src].</span>", "<span class='notice'>You slap your hands on [src].</span>")
```
Comment (TheFakeElon) [inline austation/code/game/objects/structures/tables_racks.dm:64]: \the src

File: austation/code/game/objects/structures/tables_racks.dm [hunk 2]
```diff
@@ -0,0 +1,67 @@
...
+// Table Banging
+/obj/structure/table/AltClick(mob/user)
+	// Using click cooldowns for attacks and shit here to prevent spam
+	if(user.next_move > world.time)
+		return
+	user.changeNext_move(CLICK_CD_MELEE)
+
+	if(user && Adjacent(user) && !user.incapacitated())
+		if(istype(user) && user.a_intent == INTENT_HARM)
+			user.visible_message("<span class='warning'>[user] slams [user.p_their()] palms down on [src].</span>", "<span class='warning'>You slam your palms down on [src].</span>")
```
Comment (TheFakeElon) [inline austation/code/game/objects/structures/tables_racks.dm:57]: \the src

**Review by TheFakeElon — APPROVED** [review]
just fix these and you're good to go

**Comment by MCterra10** [comment]
speedmerge

**Comment by TheFakeElon** [comment]
ayy

**Comment by TheFakeElon** [comment]
pssh

**Comment by TheFakeElon** [comment]
merge mi- oh

**Comment by amidoingitright** [comment]
smh gets merged in five minutes while my PR takes six days and a shitstorm

**Pull request merged**